### PR TITLE
Fixes the hetereogenous validators test

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Fixes the hetereogenous validators test

### JIRA ticket:
https://rchain.atlassian.net/browse/CORE-1538

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by lukasz-golebiewski** [comment]
bors try

**Comment by lukasz-golebiewski** [comment]
bors retry

File: integration-tests/test/wait.py [hunk 1]
```diff
@@ -65,6 +65,18 @@ def is_satisfied(self) -> bool:
...
+        self.node = node
+        self.block_hash = block_hash
+
+    def __str__(self) -> str:
+        args = ', '.join(repr(a) for a in (self.node.name, self.block_hash))
+        return '<{}({})>'.format(self.__class__.__name__, args)
+
+    def is_satisfied(self) -> bool:
+        block_message = self.node.get_block(self.block_hash)
+        return "Error: Failure to find block with hash" not in block_message
```
Comment (adaszko) [inline integration-tests/test/wait.py]: Hmm, I think `get_block()` should be returning non-zero exit code if something goes wrong and non-zero exit codes translate into a Python exception.

File: integration-tests/test/test_heterogenous_validators.py [hunk 2]
```diff
@@ -112,7 +111,8 @@ def test_heterogenous_validators(command_line_options_fixture, docker_client_fix
                     bonded_validator.deploy_string(forward_file)
                     bonded_validator.propose()
                     bonded_validator.deploy_string(bond_file)
-                    bonded_validator.propose()
+                    bonding_block_hash = bonded_validator.propose()
+                    wait_for_node_sees_block(context,joining_validator,bonding_block_hash)
```
Comment (adaszko) [inline integration-tests/test/test_heterogenous_validators.py]: Spaces after commas please

**Comment by lukasz-golebiewski** [comment]
bors retry

**Comment by lukasz-golebiewski** [comment]
bors retry

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
bors ping

**Comment by lukasz-golebiewski** [comment]
bors try

**Pull request merged**